src/coreclr/tools/aot/ILCompiler/ILCompiler.props

```diff
@@ -48,8 +48,7 @@
 
     <!-- CoreDisTools are used in debugging visualizers. -->
     <IncludeCoreDisTools Condition="'$(Configuration)' != 'Release' and '$(CrossHostArch)' == ''">true</IncludeCoreDisTools>
-    <!-- source-build doesn't use ObjWriter for the ILCompiler.  the end-user will end up pulling Microsoft-built bits for NativeAOT anyway.  -->
-    <IncludeObjWriter Condition="'$(DotNetBuildFromSource)' != 'true'">true</IncludeObjWriter>
+    <IncludeObjWriter Condition="'$(NativeAotSupported)' == 'true'">true</IncludeObjWriter>
```

Comment:
Do we build the ILCompiler project when NativeAotSupported != true in source build? (The original condition was source build only so I want to understand why we need this at all instead of deleting it.)

Reply:
I assumed we do because otherwise the original condition wasn't needed. I'll check.

Reply:
It looks like the project gets compiled. This is from the source-build (vmr) build log for runtime:
```
    ILCompiler.DependencyAnalysisFramework -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.DependencyAnalysisFramework/x64/Release/ILCompiler.DependencyAnalysisFramework.dll
    ILCompiler.TypeSystem -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.TypeSystem/x64/Release/ILCompiler.TypeSystem.dll
    ILCompiler.Diagnostics -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.Diagnostics/x64/Release/ILCompiler.Diagnostics.dll
    ILCompiler.ReadyToRun -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.ReadyToRun/x64/Release/ILCompiler.ReadyToRun.dll
    crossgen2 -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/coreclr/linux.x64.Release/crossgen2/crossgen2.dll
    ILCompiler.Build.Tasks -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/coreclr/linux.x64.Release/ilc-published/netstandard/ILCompiler.Build.Tasks.dll
    ILCompiler.MetadataTransform -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.MetadataTransform/x64/Release/ILCompiler.MetadataTransform.dll
    ILCompiler.Compiler -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.Compiler/x64/Release/ILCompiler.Compiler.dll
    ILCompiler.RyuJit -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/ILCompiler.RyuJit/x64/Release/ILCompiler.RyuJit.dll
    ILCompiler -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/coreclr/linux.x64.Release/ilc/ilc.dll
    Optimizing assemblies for size. This process might take a while.
    ILCompiler -> /home/tester/dotnet/src/runtime/artifacts/source-build/self/src/artifacts/bin/coreclr/linux.x64.Release/ilc-published/
```

Reply:
What I meant is that `IncludeObjWriter` was added in #76206 as a source build workaround - it allows building the ILCompiler project but it will not be functional (the compiler is unusable without the object writer and can't compile anything). I don't actually know why it was set up that way (I would have excluded the whole ILCompiler project), but it was for source build only. This is now changing the condition to something else and I think it's redundant with

https://github.com/dotnet/runtime/blob/83bf4b66ed3de5905b2e410a194a0f5f3d924c3d/eng/Subsets.props#L317

that simply excludes the whole project on platforms where we didn't bring NativeAOT up.

I think we should delete the property instead.